packages/convertkit/src/forms/subscribe/subscribe-to-convertkit-form.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react'
+import * as Yup from 'yup'
+import axios from 'axios'
+import {useFormik} from 'formik'
+import {Input, Button} from '@skillrecordings/react/dist/components'
+
+const SUBSCRIBE_API_URL =
+  process.env.NEXT_PUBLIC_SUBSCRIBE_API_URL || '/api/convertkit/subscribe'
+
+type ConvertkitSubscriber = {
+  id: string
+  email: string
+  first_name: string
+}
+
+type SubscribeFormProps = {
+  actionLabel?: string
+  successMessage?: string | React.ReactElement
+  errorMessage?: string | React.ReactElement
+  submitButtonElem?: React.ReactElement
+  onError?: (error?: any) => void
+  onSuccess?: (subscriber?: ConvertkitSubscriber) => void
+  formId: number
+  subscribeApiURL?: string
+}
+
+/**
+ * This form posts to a designated api URL (assumes /api/convertkit/subscribe
+ * by default)
+ * @param formId the Convertkit form id
+ * @param submitButtonElem an element to use as the button for the form submit
+ * @param errorMessage A string or element representing the message shown on error
+ * @param successMessage A string or element representing the message shown on success
+ * @param actionLabel Label for the button (not used if submitButtonElem is used)
+ * @param onError function to call on error
+ * @param onSuccess function to call on success
+ * @param subscribeApiURL optional param to override the api url that gets posted to
+ * @param rest anything else!
+ * @constructor
+ */
+const SubscribeToConvertkitForm: React.FC<SubscribeFormProps> = ({
+  formId,
+  submitButtonElem,
+  errorMessage = <p>Something went wrong.</p>,
+  successMessage = <p>Thanks!</p>,
+  actionLabel = 'Subscribe',
+  onError = () => {},
+  onSuccess = () => {},
+  subscribeApiURL = SUBSCRIBE_API_URL,
```

Comment:
technically, if we weren't using `email_address` you could submit to CK directly with this form, but I want to be able to specify it in the final application

Reply:
so if we want it to be `/api/list/subscribe` or whatever we could pass that in